DemoApp/ChatPresenter.swift

```diff
@@ -176,22 +176,48 @@ class DemoChatChannelListRouter: ChatChannelListRouter {
     }
 }
 
-class DemoChannelListVC: ChatChannelListVC {
+class DemoChannelListVC: ChatChannelListVC, EventsControllerDelegate {
     /// The `UIButton` instance used for navigating to new channel screen creation,
     lazy var createChannelButton: UIButton = {
         let button = UIButton()
         button.setImage(UIImage(named: "pencil")!, for: .normal)
         return button
     }()
+    
+    var eventsController: EventsController!
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
         navigationItem.rightBarButtonItem = UIBarButtonItem(customView: createChannelButton)
         createChannelButton.addTarget(self, action: #selector(didTapCreateNewChannel), for: .touchUpInside)
+        
+        eventsController = controller.client.eventsController()
+        eventsController.delegate = self
     }
 
     @objc open func didTapCreateNewChannel(_ sender: Any) {
         (router as! DemoChatChannelListRouter).showCreateNewChannelFlow()
     }
+    
+    func eventsController(_ controller: EventsController, didReceiveEvent event: Event) {
+        switch event {
+        case let event as MessageNewEvent:
+            if let message = event.message(), let channel = event.channel() {
```

Comment:
The whole approach totally makes sense to me, the only thing I'd maybe change is turn `message()`, `channel()` etc on `Event` into computed properties so when you use it the syntax is more natural, unless you think there is some value in explicitly showing that there is computation under the hood

Reply:
if those are computed properties instead of functions, we have to implement them for all conformers of `Event` type. when they're functions, we can have a default implementation and all conformers will get the default implementation